### PR TITLE
KAM26-33 Add ooma-services-notifications role to kamailio

### PR DESCRIPTION
**Pull request by enoskov-at-ooma**
Need to send SIP NOTIFY when call recording started/stopped to mobile and desktop clients. 
Refer to https://oomacorp.atlassian.net/wiki/spaces/TELEPH/pages/998375589/Shared+Hosted

### CODE REVIEW

File: kamailio/ooma-services-notifications-role.cfg [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    xlog("L_INFO", "log|get ooma inbound fax mwi, payload: $(kzE)");
+    $var(endpoint) = $(kzE{kz.json,Endpoint});
+    $var(accountdb) = $(kzE{kz.json,Account-DB});
+    $var(faxid) = $(kzE{kz.json,Msg-ID});
+    $var(callernumber) = $(kzE{kz.json,Caller-Number});
+    $var(calleenumber) = $(kzE{kz.json,Callee-Number});
+    if(reg_fetch_contacts("location", "sip:$var(endpoint)", "device")) {
+        xlog("L_INFO", "reg info 1: $(ulc(device=>aor))\n");
+        xlog("L_INFO", "reg info 2: $(ulc(device=>addr))\n");
+        xlog("L_INFO", "reg info 3: $(ulc(device=>received))\n");
```
Comment (lazedo) [inline kamailio/ooma-services-notifications-role.cfg:23]: change to `debug`

File: kamailio/ooma-services-notifications-role.cfg [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    $var(vmid) = $(kzE{kz.json,Msg-ID});
+    $var(callernumber) = $(kzE{kz.json,Caller-Number});
+    $var(callername) = $(kzE{kz.json,Caller-Name});
+    $var(length) = $(kzE{kz.json,Voicemail-Length});
+    $var(callid) = $(kzE{kz.json,Call-ID});
+    $var(vmboxid) = $(kzE{kz.json,VMBox});
+    if(reg_fetch_contacts("location", "sip:$var(endpoint)", "device")) {
+        xlog("L_INFO", "reg info 1: $(ulc(device=>aor))\n");
+        xlog("L_INFO", "reg info 2: $(ulc(device=>addr))\n");
+        xlog("L_INFO", "reg info 3: $(ulc(device=>received))\n");
```
Comment (lazedo) [inline kamailio/ooma-services-notifications-role.cfg:57]: change to `debug`

File: kamailio/ooma-services-notifications-role.cfg [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    $var(endpoint) = $(kzE{kz.json,Endpoint});
+    $var(callernumber) = $(kzE{kz.json,Caller-Number});
+    $var(calleenumber) = $(kzE{kz.json,Callee-Number});
+    $var(callid) = $(kzE{kz.json,Call-ID});
+    $var(type) = $(kzE{kz.json,Type});
+
+    if(reg_fetch_contacts("location", "sip:$var(endpoint)", "device")) {
+        xlog("L_INFO", "reg info 1: $(ulc(device=>aor))\n");
+        xlog("L_INFO", "reg info 2: $(ulc(device=>addr))\n");
+        xlog("L_INFO", "reg info 3: $(ulc(device=>received))\n");
```
Comment (lazedo) [inline kamailio/ooma-services-notifications-role.cfg:89]: change to `debug`

**Review by lazedo — COMMENTED** [review]
this seems better to just drop-in in the extras.d folder, any reason not to do that ?

also, looks like every server is receiving the payload, is this intended ?

File: kamailio/ooma-services-notifications-role.cfg [hunk 4]
```diff
@@ -0,0 +1,119 @@
+## OOMA SERVICES NOTIFICATIONS ROLE
+
+route[KZ_AMQP_BINDING_OOMA_SERVICES_NOTIFICATIONS] {
+    $var(payload) = $_s({ "exchange" : "ooma_services" , "type" : "topic", "queue" : "OOMA-SERVICES-NOTIFICATIONS-MWI-QUEUE-MY_HOSTNAME", "routing" : "notifications.infax.mwi", "auto_delete" : 1, "durable" : 0, "no_ack" : 1, "wait_for_consumer_ack" : 0, "federate" : 1, "exclusive": false});
+    kazoo_subscribe("$var(payload)");
+    $var(payload2) = $_s({ "exchange" : "ooma_services" , "type" : "topic", "queue" : "OOMA-SERVICES-NOTIFICATIONS-MWI-QUEUE-MY_HOSTNAME", "routing" : "notifications.newvm.mwi", "auto_delete" : 1, "durable" : 0, "no_ack" : 1, "wait_for_consumer_ack" : 0, "federate" : 1, "exclusive": false});
+    kazoo_subscribe("$var(payload2)");
+    $var(payload3) = $_s({ "exchange" : "ooma_services" , "type" : "topic", "queue" : "OOMA-SERVICES-NOTIFICATIONS-RECORD-MY_HOSTNAME", "routing" : "notifications.record.mwi", "auto_delete" : 1, "durable" : 0, "no_ack" : 1, "wait_for_consumer_ack" : 0, "federate" : 1, "exclusive": false});
+    kazoo_subscribe("$var(payload3)");
```
Comment (lazedo) [inline kamailio/ooma-services-notifications-role.cfg:9]: if the `exchange` is the same, you can use an array in `routing` instead
```
"routing" : ["notifications.infax.mwi", "notifications.newvm.mwi", "notifications.record.mwi"]
```

**Comment by enoskov-at-ooma** [comment]
> also, looks like every server is receiving the payload, is this intended ?

yes without subscription
in ooma_services app NOTIFY (a corresponding amqp event) is limited to mobile devices only for now

**Comment by enoskov-at-ooma** [comment]
> this seems better to just drop-in in the extras.d folder, any reason not to do that ?

ok will move to extras.d. No particular reason but let's hide it in this folder as it is very specific role needed for ooma_services.

**Comment by enoskov-at-ooma** [comment]
Review moved to https://github.com/2600hz/kazoo-configs-kamailio-extras/pull/8

**Pull request closed without merging**